com.unity.multiplayer.mlapi/Runtime/Serialization/BitWriter.cs

```diff
@@ -870,12 +870,12 @@ public void WriteByteArray(byte[] b, long count = -1)
         /// </summary>
         /// <param name="buffer"></param>
         /// <param name="targetSize"></param>
-        public void WriteBytes(byte[] buffer, long targetSize)
+        public void WriteBytes(byte[] buffer, long targetSize, int offset = 0)
         {
 
             long TargetSize = targetSize;
             long LargeInt64Blocks = (long)(TargetSize * 0.125f);
```

Comment:
This was not touched on this PR, and we discussed offline. But just for the sake of completeness, I'm feeling quite uneasy about the `*0.125`. Invoking floating-point numbers for something that could be done with integers seems risky.

Since 0.125 has an exact ieee representation, it's not so bad, but it feels like a bad habit. I'd be more at ease with `/8` or `>> 3`

Reply:
I like the right shift operation option better anyway.  Let me make that adjustment.